src/cmds/restic/cmd_find.go

```diff
@@ -192,33 +178,14 @@ func runFind(opts FindOptions, gopts GlobalOptions, args []string) error {
 		}
 	}
 
-	err = repo.LoadIndex()
-	if err != nil {
+	if err = repo.LoadIndex(); err != nil {
 		return err
 	}
 
-	pat.pattern = args[0]
-
-	if opts.CaseInsensitive {
-		pat.pattern = strings.ToLower(pat.pattern)
-		pat.ignoreCase = true
-	}
-
-	if opts.Snapshot != "" {
-		snapshotID, err := restic.FindSnapshot(repo, opts.Snapshot)
-		if err != nil {
-			return errors.Fatalf("invalid id %q: %v", args[1], err)
-		}
-
-		return findInSnapshot(repo, pat, snapshotID)
-	}
-
-	done := make(chan struct{})
-	defer close(done)
-	for snapshotID := range repo.List(restic.SnapshotFile, done) {
-		err := findInSnapshot(repo, pat, snapshotID)
-
-		if err != nil {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Use `context.TODO()` instead of `context.Background()`, the latter should only be used in the `main()` (or equivalent). The former is made specifically to allow refactoring a code base piece by piece and easily find the places where a new context has been introduced that'd need to be derived from a parent context.